### PR TITLE
AC-213 - Apply MVP - PatientDashboardActivity

### PR DESCRIPTION
**Pull request by tmarzeion**

### CODE REVIEW

**Comment by tmarzeion** [comment]
@AvijitGhosh82 @AdamGrzybkowski 
Since this PR is almost ready, it needs some manual testing to find bugs.
Tell me if You find something :)

File: openmrs-client/src/main/java/org/openmrs/mobile/adapters/SimilarPatientsRecyclerViewAdapter.java [hunk 1]
```diff
@@ -104,9 +107,22 @@ public PatientViewHolder(View itemView) {
...
+                new VisitApi().syncLastVitals(patient.getUuid());
+                ToastUtil.success("Patient with UUID " + patient.getUuid() + " is now available locally");
+            }
+            @Override
+            public void onResponse() {}
+            @Override
+            public void onErrorResponse() {
+                ToastUtil.error("Failed to fetch patient data");
+            }
+        });
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/adapters/SimilarPatientsRecyclerViewAdapter.java]: @tmarzeion This code crashes the app. You got NPE in getPatientID method due to asyncs problems probably.

File: openmrs-client/src/main/java/org/openmrs/mobile/utilities/ObservationDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                    }
+                    catch (IllegalStateException e) {
+                        observation.setDiagnosisList(diagnosisDetails.getAsJsonObject().get(VALUE_KEY).getAsString());
+                    }
+                }
+            }
+        } else if (jsonObject.get("concept") != null && "Text of encounter note".equals(jsonObject.get("concept").getAsJsonObject().get(DISPLAY_KEY).getAsString())) {
+            observation.setDiagnosisNote(jsonObject.getAsJsonObject().get(VALUE_KEY).getAsString());
+        }
+        return observation;
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/utilities/ObservationDeserializer.java]: @tmarzeion Maybe some unit tests ? :)

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/diagnosis/PatientDashboardDiagnosisPresenter.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            }
+        }
+        return diagnosis;
+    }
+
+    @Override
+    public void start() {
+        List<Encounter> mVisitNoteEncounters = new EncounterDAO().getAllEncountersByType(mPatient.getId(),
+                new EncounterType(EncounterType.VISIT_NOTE));
+        mPatientDiagnosisView.setDiagnosesToDisplay(getAllDiagnosis(mVisitNoteEncounters));
```
Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/diagnosis/PatientDashboardDiagnosisPresenter.java]: Fixed

File: openmrs-client/src/main/java/org/openmrs/mobile/adapters/SimilarPatientsRecyclerViewAdapter.java (hunk 1 above)
Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/adapters/SimilarPatientsRecyclerViewAdapter.java]: Fixed

File: openmrs-client/src/main/java/org/openmrs/mobile/utilities/ObservationDeserializer.java (hunk 2 above)
Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/utilities/ObservationDeserializer.java]: Done:
- ObservationDeserializerTest.java

**Pull request merged**